storage/flux/reader.go

```diff
@@ -619,11 +620,20 @@ func (wai *windowAggregateIterator) Do(f func(flux.Table) error) error {
 func (wai *windowAggregateIterator) handleRead(f func(flux.Table) error, rs storage.ResultSet) error {
 	windowEvery := wai.spec.WindowEvery
 	createEmpty := wai.spec.CreateEmpty
+
+	selector := false
+	if len(wai.spec.Aggregates) > 0 {
```

Comment:
Can you change this entire conditional to a function?

```go
func isSelector(kind plan.ProcedureKind) bool {
}
```

You can either continue to do the conditional in the same way, make a gigantic conditional, or a switch statement. It will help to declutter this method and clearly communicate that we are looking for a selector.